internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -439,3 +474,16 @@ func reconcileExternalTemplateReference(ctx context.Context, c client.Client, cl
 
 	return patchHelper.Patch(ctx, obj)
 }
+
+func descendantMachineSets(objs []*clusterv1.MachineSet) string {
```

Comment:
```suggestion
func machineSetsNames(objs []*clusterv1.MachineSet) string {
```
Note: ideally this can become a utils that works with all client.Objects, but this is something we can consider also as a follow up

Reply:
added a util, not sure if the place where I added it is the best.